src/client.ts

```diff
@@ -545,6 +537,50 @@ export class StreamChat<
     this._user = { ...user };
   }
 
+  /**
+   * Disconnects the websocket connection, without removing the user set on client.
+   * client.closeConnection will not trigger default auto-retry mechanism for reconnection. You need
+   * to call client.reconnectUser to reconnect to websocket.
+   *
+   * This is mainly useful on mobile side. You can only receive push notifications
+   * if you don't have active websocket connection.
+   * So when your app goes to background, you can call `client.closeConnection`.
+   * And when app comes back to foreground, call `client.openConnection`.
+   *
+   * @param timeout Max number of ms, to wait for close event of websocket, before forcefully assuming succesful disconnection.
+   *                https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent
+   */
+  closeConnection = (timeout?: number) => {
+    if (this.cleaningIntervalRef != null) {
+      clearInterval(this.cleaningIntervalRef);
+      this.cleaningIntervalRef = undefined;
+    }
+
+    for (const channel of Object.values(this.activeChannels)) {
+      channel._disconnect();
+    }
+
+    return this.wsConnection?.disconnect(timeout);
+  };
+
+  /**
+   * Reconnects the current user.
+   */
+  openConnection = () => {
```

Comment:
I think here we should check for any existing connection and either throw an error or just return the active WS

Reply:
We do that inside connection class.